### PR TITLE
[MM-12333] Convert mocha to jest and move test files adjacent to the files being tested

### PR DESCRIPTION
**Pull request by saturninoabril**
#### Summary
1. Converted test runner from Mocha to Jest
- removed all mocha and related packages
- only consider `console.error` and `console.warn` logs to fail during test, and let the `console.log` be ignored since we're using `console.log` in several places including the redux library (e.g. in retrying an action).
- test for server compatibility of getTeammateNameDisplaySetting is broken.  I just removed it and kept the current server test only. I think there's no point of keeping that test.
- downgrade `console.warn` to `console.log` of scheme's scope.  This is to prevent overall test failures after checking all console errors and warnings.

2.  Moved test files adjacent to the files being tested.

Note:  I added all team leads to make sure all are aware on this change.

#### Ticket Link
Jira ticket: [MM-12333](https://mattermost.atlassian.net/browse/MM-12333)

#### Checklist
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit tests passed
- [x] Ran `make flow` to ensure type checking passed
- [x] Added or updated unit tests (required for all new features)

### CODE REVIEW

**Comment by enahum** [comment]
@cpanato @lieut-data any idea why this PR is failing?
image

File: package.json [hunk 1]
```diff
@@ -73,7 +73,26 @@
...
+    "coverageReporters": [
+      "lcov",
+      "text-summary"
+    ],
+    "setupTestFrameworkScriptFile": "<rootDir>/test/setup.js",
+    "testPathIgnorePatterns": [
+      "/node_modules/"
+    ],
+    "moduleNameMapper": {
+      "assets/images/video_player/(.*).png": "<rootDir>/dist/assets/images/video_player/$1@2x.png"
```
Comment (enahum) [inline package.json]: why do we have images here?

**Comment by cpanato** [comment]
Looks like this was removed:
/test/mocha-multi-reporters-config.json
```
{
    "reporterEnabled": "spec, mocha-junit-reporter",
    "mochaJunitReporterReporterOptions": {
        "mochaFile": "./build/test-results.xml"
    }
}
```
and now the pipeline is failing because of that. Or we add this back or some other thing to get the test results

or we remove this:
https://github.com/mattermost/mattermost-redux/blob/master/build/Jenkinsfile#L38

**Comment by lieut-data** [comment]
@enahum & @saturninoabril, you can copy the webapp configuration for this, previously setup for Jest:

https://github.com/mattermost/mattermost-webapp/blob/9723e312b2ac89f47b5799c769be7cd211ab1ae3/build/Jenkinsfile#L30-L39

https://github.com/mattermost/mattermost-webapp/blob/9723e312b2ac89f47b5799c769be7cd211ab1ae3/package.json#L165-L167

**Comment by saturninoabril** [comment]
Thanks @lieut-data and @cpanato, now updated.

**Comment by jwilander** [comment]
@saturninoabril can you post an announcement in the Developers and Redux channels on community letting everyone know it's happening before you merge this?

**Comment by saturninoabril** [comment]
@jwilander I already posted at those channels
- [dev](https://community.mattermost.com/core/pl/d3nufyk5mjya9naioabtzg4cno)
- [redux](https://community.mattermost.com/core/pl/ockp337z6by47przyz8cn4o36a)

I'll comment on that posts once this is ready to merge.

**Comment by saturninoabril** [comment]
Rebased and fixed merge conflicts.

@enahum Please take a look again, thanks!

**Pull request merged**